lib/diff/lcs/ldiff.rb

```diff
@@ -99,74 +108,64 @@ def run(args, _input = $stdin, output = $stdout, error = $stderr) # :nodoc:
     # items we've read from each file will differ by FLD (could be 0).
     file_length_difference = 0
 
-    data_old = File.read(file_old)
-    data_new = File.read(file_new)
-
     # Test binary status
     if @binary.nil?
       old_bin = data_old[0, 4096].include?("\0")
       new_bin = data_new[0, 4096].include?("\0")
       @binary = old_bin || new_bin
     end
```

Comment:
```suggestion
    if @binary.nil?
      old_bin = info_old.data[0, 4096].include?("\0")
      new_bin = info_new.data[0, 4096].include?("\0")
      @binary = old_bin || new_bin
    end
```

Reply:
Oops, it seems I improperly rebased on top of #104, fixing it!